e2e/test/scenarios/models/reproductions.cy.spec.js

```diff
@@ -1597,51 +1597,49 @@ describe("issue 32963", () => {
 });
 
 describe("issues 35039 and 37009", () => {
+  // We only need to ensure there is a comment. Any comment.
+  const query = "select * from products limit 1 -- foo";
+
+  const cardDetails = {
+    name: "35039",
+    type: "model",
+    native: { query },
+    visualization_settings: {},
+  };
+
   beforeEach(() => {
     restore();
     cy.intercept("POST", "/api/dataset").as("dataset");
     cy.signInAsNormalUser();
+
+    createNativeQuestion(cardDetails).then(({ body: { id } }) => {
+      // It is crucial for this repro to go directly to the "edit query definition" page!
+      // When the repro was created back in v47-v48, it was still possible to save a new model
+      // without running the query first. This resulted in the missing `result_metadata`.
+      // It's not possible to replicate that using UI anymore, so our best bet is to create a model
+      // using API, and then to visit this page directly.
+      cy.visit(`/model/${id}/query`);
+    });
+    assertResultsLoaded();
   });
 
   // This test follows #37009 repro steps because they are simpler than #35039 but still equivalent
   it("should show columns available in the model (metabase#35039) (metabase#37009)", () => {
-    cy.visit("/model/new");
-    cy.findByTestId("new-model-options")
-      .findByText("Use a native query")
-      .click();
-
-    focusNativeEditor().type("select * from products -- where true=false");
-    cy.findByTestId("native-query-editor-container").icon("play").click();
-    cy.wait("@dataset");
-
-    cy.findByTestId("dataset-edit-bar").button("Save").click();
-    modal()
-      .last()
-      .within(() => {
-        cy.findByLabelText("Name").type("Model").realPress("Tab");
-        cy.findByText("Save").click();
-      });
-
-    openQuestionActions();
-    popover().findByText("Edit query definition").click();
-
-    focusNativeEditor().type(
-      "{backspace}{backspace}{backspace}{backspace}{backspace}",
-    );
+    // The repro requires that we update the query in any way while still preserving the comment.
```

Comment:
This and the first comment in this `describe` suggest that it's about comments.
We can be more specific and explain that we need to make a change to the query but without making change to query results. Editing a comment is just one of ways of doing that.

Reply:
This was based on Zoltan's key for the reproduction.

> The query builder does not show source fields for models based on native queries, *when the query has trailing comments*.

I understood that the comment is crucial.
Did I get it wrong?

Reply:
It's definitely not comment-specific. Also see #37009 for this comment:

> Update the query definition of model2, **and make a minor, non-impactful change**, for instance:

Reply:
https://github.com/metabase/metabase/pull/46038/commits/6f9aba44aab641b90388707c2a799be2b44e1f80